iroh-p2p/src/node.rs

```diff
@@ -543,6 +554,51 @@ impl<KeyStorage: Storage> Node<KeyStorage> {
                         QueryResult::Bootstrap(Err(e)) => {
                             warn!("kad bootstrap error: {:?}", e);
                         }
+                        QueryResult::GetClosestPeers(Ok(GetClosestPeersOk { key, peers })) => {
+                            debug!("GetClosestPeers ok {:?}", key);
+                            if let Some(KadQueryChannel::FindPeerOnDHT {
+                                channels, peer_id, ..
+                            }) = self.kad_queries.remove(&QueryKey::FindPeerOnDHTKey(key))
+                            {
+                                if !peers.contains(&peer_id) {
```

Comment:
Can this be DRYed? I know anyhow::Error is not clone, so it is not a matter of just storing the result and then sending out clones. But maybe with a closure?

```rust
let have_peer = peers.contains(&peer_id)
let res = || if have_peer { Ok(()) } else { Err(anyhow!("nope")) }
tokio::task::spawn(async move {
    for chan in channels.into_iter() {
        chan.send(res()).ok();
    }
});
```

Reply:
This is slick! `Error` not being `clone` was the exact issue I ran into here & I wasn't nimble enough to figure out how to make it DRY. Very much appreciated.